source/_integrations/zha.markdown

```diff
@@ -138,24 +142,13 @@ Click on **ADD DEVICES** to start a scan for new devices.
 
 Reset your Zigbee devices according to the device instructions provided by the manufacturer (e.g.,  turn on/off lights up to 10 times, switches usually have a reset button/pin).
 
-## Troubleshooting
-
-### Add Philips Hue bulbs that have previously been added to another bridge
-
-Philips Hue bulbs that have previously been added to another bridge won't show up during search. You have to restore your bulbs back to factory settings first. To achieve this, you basically have the following options.
-
-#### Philips Hue Dimmer Switch
-
-Using a Philips Hue Dimmer Switch is probably the easiest way to factory-reset your bulbs. For this to work, the remote doesn't have to be paired with your previous bridge.
+## ZHA exception and deviation handling
 
-1. Turn on your Hue bulb you want to reset
-2. Hold the Dimmer Switch near your bulb (< 10 cm)
-3. Press and hold the (I)/(ON) and (O)/(OFF) buttons of the Dimmer Switch for about 10 seconds until your bulb starts to blink
-4. Your bulb should stop blinking and eventually turning on again. At the same time, a green light on the top left of your remote indicates that your bulb has been successfully reset to factory settings.
+Zigbee devices that deviate from or do not fully conform to the standard specifications set by the [Zigbee Alliance](https://zigbeealliance.org) may require the development of custom [ZHA Device Handlers](https://github.com/dmulcahey/zha-device-handlers) (ZHA custom quirks handler implementation) to for all their functions to work properly with the ZHA integration in Home Assistant. These ZHA Device Handlers for Home Assistant can thus be used to parse custom messages to and from Zigbee devices.
 
-#### hue-thief
+The custom quirks implementations for zigpy implemented as ZHA Device Handlers for Home Assistant are a similar concept to that of [Hub-connected Device Handlers for the SmartThings Classics platform](https://docs.smartthings.com/en/latest/device-type-developers-guide/) as well as that of [Zigbee-Shepherd Converters as used by Zigbee2mqtt](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html), meaning they are each virtual representations of a physical device that expose additional functionality that is not provided out-of-the-box by the existing integration between these platforms.
```

Comment:
I think this paragraph can be removed as a whole.

Reply:
If you mean the paragraph about "ZHA exception and deviation handling" then I disagree, creating custom quirks (ZHA Device Handlers) for zhaquirks is something advanced Home Assistant end-users can do themselves in order to add support for devices that do not follow the standards. This is also something that is Home Assistant specific and not really genric to the zigpy library as I'm sure dmulcahey can attest to . Please see: https://github.com/dmulcahey/zha-device-handlers

Reply:
An average will have no clue on what this means even...

Reply:
Maybe but it is still specific to this ZHA integration in Home Assistant. Perhaps it should just be clarified that "advanced users" can add support for not working devices via ZHA Device Handlers?

Reply:
Accept to resolve this as a moot point for PR as that paragraph was just moved further down?

Reply:
@dmulcahey & @Adminiuga What are your opinion on keeping info on ZHA Device Handlers?

Reply:
You are trying to raise a discussion in a resolved comment. Please unresolve it again in that case. We get the notification, but your comment isn't visible if the conversation is resolved. I've now unresolved it.